docs/tutorials/build/formula.md

```diff
@@ -0,0 +1,653 @@
+---
+title: Basic formula
+hide:
+- toc
+---
+
+# Learn to build a formula
+
+Formulas are one of the most basic building blocks in Coda; used to calculate values, filter tables, and so much more. In this tutorial you'll learn how to build a basic formula that reverses text.
+
+Before starting this tutorial, make sure you have completed one of the **Get started** tutorials, either [In your browser][quickstart_web] or [On your local machine][quickstart_cli].
+
+
+## 📐 Design the formula
+
+Before we start writing any code, let's first think through how we want the formula to look and work. This includes the name, the inputs, the outputs, etc. For a formula that reverses text we could imagine it working like this:
+
+```
+Reverse("Hello") ==> "olleH"
+```
+
+It's also a good time think through how it should handle edge cases, where the input is different than what we'd normally expect. For example, empty inputs, large amounts of data, etc.
+
+```
+Reverse("a") ==> "a"
+Reverse("") ==> ""
+```
+
+Now that we have a destination in mind, let's get coding!
+
+
+## 🏗️ Scaffold the formula
+
+We'll start by scaffolding out the structure of the formula. Once we have it looking the way we want we can go back and add the logic later.
+
+### Add the Pack boilerplate
+
+<section class="tutorial-row" markdown>
+<div markdown>
+
+For any new Pack you need to start with the boilerplate code that sets it up. This includes an import statement which loads the Pack SDK library, and the creation of our Pack definition.
+
+</div>
+<div markdown>
+
+```ts
+import * as coda from "@codahq/packs-sdk";
+export const pack = coda.newPack();
+```
+
+</div>
+</section>
+
+
+### Add the formula definition
+
+<section class="tutorial-row" markdown>
+<div markdown>
+
+Next add a formula definition to your Pack using the `addFormula()` method. This method takes in a set of [key-value pairs][javascript_key_value] that configure the various settings of the formula.
+
+</div>
+<div markdown>
+
+```{.ts hl_lines="4-6"}
+import * as coda from "@codahq/packs-sdk";
+export const pack = coda.newPack();
+
+pack.addFormula({
+
+});
+```
+
+</div>
+</section>
+
+### Set the name and description
+
+<section class="tutorial-row" markdown>
+<div markdown>
+
+The first key-value pairs to set are the name and description of the formula. The name determines how users will call it in the Coda formula language, and the description is help text users can refer to.
+
+</div>
+<div markdown>
+
+```{.ts hl_lines="5-6"}
+import * as coda from "@codahq/packs-sdk";
+export const pack = coda.newPack();
+
+pack.addFormula({
+  name: "Reverse",
+  description: "Reverses some text.",
+});
+```
+
+</div>
+</section>
+
+
+### Create the parameters array
```

Comment:
Q: are using "array" everywhere rather than "list"?

Reply:
I've been using "list" to refer to the Coda type (`List()`, etc) or conceptual lists ("Generate the URL list"). In this step the user is specifically creating an empty JS array, to be filled in later, so I felt like "array" was a better fit.